### PR TITLE
argparse_help_cb with no exit

### PR DESCRIPTION
**Pull request by a-random-lemurian**
This can be used to print more information that simply cannot be placed in the `epilogue` param of argparse_describe(), such as licensing info or subcommand info

### CODE REVIEW

**Comment by mschilli87** [comment]
How would I use this in practice? It is not called anywhere and not exported via the header file. I think some documentation would should be added before merging this.

File: argparse.c [hunk 1]
```diff
@@ -384,3 +384,12 @@ argparse_help_cb(struct argparse *self, const struct argparse_option *option)
     argparse_usage(self);
```
Comment (mschilli87) [inline argparse.c:384]:
```suggestion
argparse_help_cb_no_exit(self, option);
```
Just to avoid the code duplication and make both versions to stay in sync.

Comment (mschilli87) [reply argparse.c:384]: PS: I did select two lines before posting the suggestion but it seems GitHub doesn't support this (contrary to [this announcement](https://github.blog/changelog/2020-02-26-multi-line-code-suggestions-beta/)). So please make sure to also drop the `void(option);` line from `argparse_help_cb` when applying this change.

Comment (a-random-lemurian) [reply argparse.c:384]: I decided to split it into two separate functions to maintain backwards compatibility.

Comment (a-random-lemurian) [reply argparse.c:384]: Oh, I forgot to put it in the header file. Oops...

Comment (mschilli87) [reply argparse.c:384]: I don't understand that point: If `argparse_help_cb_no_exit` is identical to `argparse_help_cb` expect for the last line then how would calling it there break backwards compatibilty. Clearly I miss something here because I don't get the point of introducing `argparse_help_cb_no_exit` without using or exporting it in the first place.

Comment (a-random-lemurian) [reply argparse.c:384]: > I don't understand that point: If `argparse_help_cb_no_exit` is identical to `argparse_help_cb` expect for the last line then how would calling it there break backwards compatibilty? Clearly I miss something here because I don't get the point of introducing `argparse_help_cb_no_exit` without using or exporting it in the first place.

Existing code would likely be broken by the new argument if we were not to duplicate the function.

File: argparse.h [hunk 2]
```diff
@@ -124,6 +126,7 @@ void argparse_describe(struct argparse *self, const char *description,
 int argparse_parse(struct argparse *self, int argc, const char **argv);
 void argparse_usage(struct argparse *self);
 
+
```
Comment (mschilli87) [inline argparse.h:129]: Was this intended? Looks unrelated to the commit to me. So if this line is missing to satisfy some coding style or so, I'd suggest putting it in a separate commit to allow reverting the introduction of `argparse_help_cb_no_exit` (if needed) without undoing this change.

Comment (a-random-lemurian) [reply argparse.h:129]: This was unintended.

File: argparse.c (hunk 1 above)
Comment (mschilli87) [reply argparse.c:384]: What new argument?

Do make it clear, I suggest replacing

```c
int
argparse_help_cb(struct argparse *self, const struct argparse_option *option)
{
    (void)option;
    argparse_usage(self);
    exit(EXIT_SUCCESS);
}

int
argparse_help_cb_no_exit(struct argparse *self,
                         const struct argparse_option *option)
{
    (void)option;
    argparse_usage(self);
}
```

by

```c
int
argparse_help_cb_no_exit(struct argparse *self,
                         const struct argparse_option *option)
{
    (void)option;
    argparse_usage(self);
}

int
argparse_help_cb(struct argparse *self, const struct argparse_option *option)
{
    argparse_help_cb_no_exit(self, option)
    exit(EXIT_SUCCESS);
}
```

Comment (a-random-lemurian) [reply argparse.c:384]: LGTM.. Let me `git rebase -i` real quick.

File: argparse.c [hunk 3]
```diff
@@ -379,8 +379,16 @@ argparse_usage(struct argparse *self)
...
 int
 argparse_help_cb(struct argparse *self, const struct argparse_option *option)
+{
+    argparse_help_cb_no_exit(self, option);
+    exit(EXIT_SUCCESS);
+}
+
+int
+argparse_help_cb_no_exit(struct argparse *self,
+                         const struct argparse_option *option)
```
Comment (mschilli87) [inline argparse.c:389]: It's a matter of style but I'd prefer to implement the calling function before the called one. Just because most people tend to read from top to bottom so IMHO it helps to already know what `argparse_help_cb_no_exit` does when seeing it being called in `argparse_help_cb`.

File: argparse.c [hunk 4]
```diff
@@ -379,8 +379,16 @@ argparse_usage(struct argparse *self)
...
 {
     (void)option;
     argparse_usage(self);
-    exit(EXIT_SUCCESS);
 }
+
```
Comment (mschilli87) [inline argparse.c:394]: This looks like another unintended change.

File: argparse.h (hunk 2 above)
Comment (mschilli87) [reply argparse.h:129]: Then I suggest dropping it from this commit. :wink:

**Review by mschilli87 — COMMENTED** [review]
Just a few nitpicks left from my side. And remember to [assign @cofyc](https://github.com/cofyc/argparse/pull/55#issuecomment-1036124186) when you think this is ready.

**Comment by cofyc** [comment]
/approve
Thanks!
